agent/http_decode_test.go

```diff
@@ -0,0 +1,3304 @@
+	},
+	{
+		desc:    "not a map (int)",
+		in:      `1`,
+		wantErr: true,
+	},
+}
+
+// =======================================================
+// TranslateKeys:
+// =======================================================
+type translateKeyTestCase struct {
+	jsonFmtStr string
+	desc       string
+	in         []interface{}
+	want       interface{}
+	equalityFn func(outStruct, wantVal interface{}) error
+}
+
+// FixupCheckType's Translate Keys:
+// 	lib.TranslateKeys(rawMap, map[string]string{
+// 	"args":                              "ScriptArgs",
+// 	"script_args":                       "ScriptArgs",
+// 	"deregister_critical_service_after": "DeregisterCriticalServiceAfter",
+// 	"docker_container_id":               "DockerContainerID",
+// 	"tls_skip_verify":                   "TLSSkipVerify",
+// 	"service_id":                        "ServiceID",
+
+var translateCheckTypeTCs = [][]translateKeyTestCase{
+	translateScriptArgsTCs,
+	translateDeregisterTCs,
+	translateDockerTCs,
+	translateTLSTCs,
+	translateServiceIDTCs,
+}
+
+// ScriptArgs: []string
+func scriptArgsEqFn(out interface{}, want interface{}) error {
+	var got []string
+	switch v := out.(type) {
+	case structs.CheckDefinition:
+		got = v.ScriptArgs
+	case *structs.CheckDefinition:
+		got = v.ScriptArgs
+	case structs.CheckType:
+		got = v.ScriptArgs
+	case *structs.CheckType:
+		got = v.ScriptArgs
+	case structs.HealthCheckDefinition:
+		got = v.ScriptArgs
+	case *structs.HealthCheckDefinition:
+		got = v.ScriptArgs
+	default:
+		panic(fmt.Sprintf("unexpected type %T", out))
+	}
+
+	wantSlice := want.([]string)
+
+	if len(got) != len(wantSlice) {
+		return fmt.Errorf("ScriptArgs: expected %v, got %v", wantSlice, got)
+	}
+	for i := range got {
+		if got[i] != wantSlice[i] {
+			return fmt.Errorf("ScriptArgs: [i=%d] expected %v, got %v", i, wantSlice, got)
+		}
+	}
+	return nil
+}
+
+var scriptFields = []string{
+	`"ScriptArgs": %s`,
+	`"args": %s`,
+	`"script_args": %s`,
+}
+
+var translateScriptArgsTCs = []translateKeyTestCase{
+	{
+		desc:       "scriptArgs: all set",
+		in:         []interface{}{`["1"]`, `["2"]`, `["3"]`},
+		want:       []string{"1"},
+		jsonFmtStr: "{" + strings.Join(scriptFields, ",") + "}",
+		equalityFn: scriptArgsEqFn,
+	},
+	{
+		desc:       "scriptArgs: first and second set",
+		in:         []interface{}{`["1"]`, `["2"]`},
+		want:       []string{"1"},
+		jsonFmtStr: "{" + scriptFields[0] + "," + scriptFields[1] + "}",
+		equalityFn: scriptArgsEqFn,
+	},
+	{
+		desc:       "scriptArgs: first and third set",
+		in:         []interface{}{`["1"]`, `["3"]`},
+		want:       []string{"1"},
+		jsonFmtStr: "{" + scriptFields[0] + "," + scriptFields[2] + "}",
+		equalityFn: scriptArgsEqFn,
+	},
+	// {
+	// 	desc:       "scriptArgs: second and third set",
+	// 	in:         []interface{}{`["2"]`, `["3"]`},
+	// 	want:       []string{"2"},
+	// 	jsonFmtStr: "{" + scriptFields[1] + "," + scriptFields[2] + "}",
+	// 	equalityFn: scriptArgsEqFn,
+	// },
```

Comment:
Is this case invalid?

Reply:
ooh yes; for this case (and seemingly only this one), sometimes the value comes up `["2"]` and sometimes `["3"]`. I assumed this is because of map ordering / iterating through a map will not always give the same order. But I also assumed that this case shouldn't matter much as it's unlikely that an API consumer would send both `"args"` & `"script_args"` simultaneously.

I figured I would uncomment it once I've got the implementation in. I'll add a TODO comment though. Does that sound OK?